src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,990 @@
+     * @param string $ofPath   The other path.
+     *
+     * @return bool Whether the base path is a base path of the other path.
+     *
+     * @since 1.0 Added method.
+     * @since 2.0 Method now fails if $basePath or $ofPath is not a string.
+     */
+    public static function isBasePath(string $basePath, string $ofPath): bool
+    {
+        $basePath = self::canonicalize($basePath);
+        $ofPath = self::canonicalize($ofPath);
+
+        // Append slashes to prevent false positives when two paths have
+        // a common prefix, for example /base/foo and /base/foobar.
+        // Don't append a slash for the root "/", because then that root
+        // won't be discovered as common prefix ("//" is not a prefix of
+        // "/foobar/").
+        return 0 === strpos($ofPath.'/', rtrim($basePath, '/').'/');
+    }
+
+    /**
+     * Splits a part into its root directory and the remainder.
+     *
+     * If the path has no root directory, an empty root directory will be
+     * returned.
+     *
+     * If the root directory is a Windows style partition, the resulting root
+     * will always contain a trailing slash.
+     *
+     * list ($root, $path) = Path::split("C:/webmozart")
+     * // => array("C:/", "webmozart")
+     *
+     * list ($root, $path) = Path::split("C:")
+     * // => array("C:/", "")
+     *
+     * @param string $path The canonical path to split.
+     *
+     * @return string[] An array with the root directory and the remaining
+     *                  relative path.
+     */
+    private static function split(string $path): array
+    {
+        if ('' === $path) {
+            return ['', ''];
+        }
+
+        // Remember scheme as part of the root, if any
+        if (false !== ($pos = strpos($path, '://'))) {
+            $root = substr($path, 0, $pos + 3);
+            $path = substr($path, $pos + 3);
+        } else {
+            $root = '';
+        }
+
+        $length = strlen($path);
+
+        // Remove and remember root directory
+        if (strpos($path, '/') === 0) {
+            $root .= '/';
+            $path = $length > 1 ? substr($path, 1) : '';
+        } elseif ($length > 1 && ctype_alpha($path[0]) && ':' === $path[1]) {
+            if (2 === $length) {
+                // Windows special case: "C:"
+                $root .= $path.'/';
+                $path = '';
+            } elseif ('/' === $path[2]) {
+                // Windows normal case: "C:/"..
+                $root .= substr($path, 0, 3);
+                $path = $length > 3 ? substr($path, 3) : '';
+            }
+        }
+
+        return [$root, $path];
+    }
+
+    /**
+     * Converts string to lower-case (multi-byte safe if mbstring is installed).
+     *
+     * @param string $str The string
+     *
+     * @return string Lower case string
+     */
+    private static function toLower(string $str): string
+    {
+        if (function_exists('mb_strtolower')) {
```

Comment:
I suggest depending on the mbstring polyfill and always using `mb_strtolower`, as we do elsewhere in Symfony.